### DISCUSSION TITLE
[V1] Why there is no swap queue in V1 version Scheduler?

### DISCUSSION THREAD

**Discussion by Ghjk94522** [vllm-project/vllm#11082] [Q&A]
As mentioned in the title, there is no swap deque in class Scheduler, and no swap relative operation in schedule func.
Is this not implemented yet or just remove for some reason?

```python
class Scheduler:
    def __init__(
        self,
        scheduler_config: SchedulerConfig,
        cache_config: CacheConfig,
        lora_config: Optional[LoRAConfig],
    ) -> None:
        ...
        self.requests: Dict[str, Request] = {}
        # Priority queues for requests.
        self.waiting: Deque[Request] = deque()
        self.running: List[Request] = []
        ...
```

codes in the vllm/v1/core/scheduler.py shown as above

And I found the `_initialize_kv_caches` func in class EngineCore init the cpu blocks number to 0 as followed:

```python
 def _initialize_kv_caches(self,
                              cache_config: CacheConfig) -> Tuple[int, int]:
        num_gpu_blocks, _ = self.model_executor.determine_num_available_blocks(
        )
        ...
        num_cpu_blocks = 0
        self.model_executor.initialize(num_gpu_blocks)
        return num_gpu_blocks, num_cpu_blocks
```

Maybe someone can help me to explain this?

**Comment by comaniac** [comment]
Swapping in v1 is removed intentionally. Instead, we expect when recomputing the preempted requests, most prompt tokens could be bypassed due to prefix caching.

**Reply by Ghjk94522** [reply]
get, thanks
